flight/Modules/Actuator/actuator.c

```diff
@@ -413,37 +414,56 @@ float ProcessMixer(const int index, const float curve1, const float curve2,
 	return(result);
 }
 
-
 /**
- *Interpolate a throttle curve. Throttle input should be in the range 0 to 1.
- *Output is in the range 0 to 1.
+ * Interpolate a mixer curve. Input is assumed to be in the range [0,1] unless otherwise specified.
+ * Output range is defined by the curve vector
+ * curve is defined in N intervals connecting N+1 points
+ * throttle curve assumes input is [0,1]
+ * this means that the throttle channel neutral value is nearly the same as its min value
+ * this is convenient for throttle, since the neutral value is used as a failsafe and would thus shut off the motor
+ * however, this makes no sense for collective, since the collective neutral point is not the min point for most vehicles with collective pitch
+ * instead, we need to accept input in [-1,1] so that the neutral point may be set arbitrarily within the typical channel input range, which is [-1,1]
  */
-static float MixerCurve(const float throttle, const float* curve, uint8_t elements)
+static float MixerCurve( float const input, float const * curve, uint8_t elements, const float input_min, const float input_max )
 {
-	float scale = throttle * (float) (elements - 1);
+	// shift our input [min,max] into the typical range [0,1]
+	float scale = ( ( input - input_min ) / ( input_max - input_min ) ) * (float) (elements - 1);
```

Comment:
Yes, the compiler is usually smart enough.  It will fix it up if inlined, or if it notices a given variant set of args is often provided within a module, it'll generate a .constprop.n symbol that has performed constant propagation.